packages/quantic/force-app/custom-token/main/default/classes/CustomTokenProvider.cls

```diff
@@ -0,0 +1,119 @@
+/**
+ * Example class showing off how to create your own TokenProvider,
+ * using a [CustomSettings](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_custom_settings.htm)
+ * to store the sensitive data such as the API Key.
+ */
+global with sharing class CustomTokenProvider implements ITokenProvider {
+    global with sharing class HttpException extends Exception {}
+    global with sharing class InvalidTokenSettingsException extends Exception {}
+
+    private static final String CONTENT_TYPE_APPLICATION_JSON = 'application/json; charset=UTF-8';
+    private static final String TOKEN_PATH = '/rest/search/v2/token';
+
+    private static Boolean isValidURL(String candidateURL) {
+        if(String.isEmpty(candidateURL)) {
+            return false;
+        }
+        try {
+            new Url(candidateURL);
+        } catch (Exception e) {
+            return false;
+        }
+        return true;
+    }
+
+    private static Map<String,String> getCurrentUserIdentity() {
+        Map<String,String> currentUserInfo = new Map<String,String>();
+        currentUserInfo.put('name', UserInfo.getUserEmail());
+        currentUserInfo.put('type', 'User');
+        currentUserInfo.put('provider', 'Email Security Provider');
+        return currentUserInfo;
+    }
+
+    /**
+     * Return the array of user identities to be concerned by the token.
+     * See https://docs.coveo.com/en/56/build-a-search-ui/search-token-authentication#userids-array-of-restuserid-required.
+     */
+    private static List<Map<String,String>> getUserIdentities() {
+        List<Map<String,String>> userIdentities = new List<Map<String,String>>();
+        userIdentities.add(CustomTokenProvider.getCurrentUserIdentity());
+        return userIdentities;
+    }
+
+    /**
+     * Ensure that all the field of the @see `TokenSettings__c` are not empty and that the URL of the platform is actually an URL.
+     * @param {TokenSettings__c} settings the settings to check.
+     * @return {Boolean} false if the settings are valid. True otherwise.
+     */
+    private static Boolean isTokenSettingInvalid(TokenSettings__c settings) {
```

Comment:
It is kind of counter-intuitive to have a method returning `true` when data is invalid. I guess `isTokenSettingValid` would be more natural.